### PR TITLE
Generate types for config-components props

### PR DESCRIPTION
**Pull request by const314**

### CODE REVIEW

File: packages/devextreme-react-generator/src/generator.ts [hunk 1]
```diff
@@ -44,12 +47,84 @@ enum BaseTypes {
...
+function isObjectDescriptor(typeDescriptor: ITypeDescr): typeDescriptor is IObjectDescr {
+  return typeDescriptor.type === 'Object' && (typeDescriptor as IObjectDescr).fields !== undefined;
+}
+
+export function convertToBaseType(type: string): BaseTypes {
+  return BaseTypes[type];
+}
+
+export function getComplexOptionType(types: ITypeDescr[]): string | undefined {
+  const typeCount = types.length;
```
Comment (IlyaKhD) [inline packages/devextreme-react-generator/src/generator.ts:72]: **Minor**
```suggestion
  const typesCount = types.length;
```

File: packages/devextreme-react-generator/src/generator.ts [hunk 2]
```diff
@@ -44,12 +47,84 @@ enum BaseTypes {
...
+
+    if (isArrayDescriptor(typeDescriptor)) {
+      return formatArrayDescriptor(typeDescriptor);
+    }
+    if (isFunctionDescriptor(typeDescriptor)) {
+      const result = formatFunctionDescriptor(typeDescriptor);
+      if (typeCount > 1) {
+        // TS1385
+        return `(${result})`;
+      }
```
Comment (IlyaKhD) [inline packages/devextreme-react-generator/src/generator.ts:105]: **Minor**
```suggestion

      // TS1385
      return `(${result})`;
```

**Review by IlyaKhD — APPROVED** [review]

File: packages/devextreme-react-generator/src/generator.ts (hunk 1 above)
Comment (const314) [reply packages/devextreme-react-generator/src/generator.ts]: fixed

File: packages/devextreme-react-generator/src/generator.ts (hunk 2 above)
Comment (const314) [reply packages/devextreme-react-generator/src/generator.ts]: fixed

**Review by IlyaKhD — APPROVED** [review]

**Review by IlyaKhD — APPROVED** [review]

**Pull request merged**